### ISSUE TITLE
using typescript definitions?

### ISSUE THREAD

**Issue by deostroll** [konvajs/konva#144]
Need guidance about using the typescript definitions. I am aware it isn't available on [DefinitelyTyped](https://github.com/konvajs/konva/issues/91), and I happen to notice you do have a **.d.ts** file in the repo.

Please provide steps about how to correctly consume this file in a typescript project

**Comment by Garee** [comment]
Reference the file at the top of each file that uses the definitions.

```typescript
/// <reference path="path/to/konva/resources/konva.d.ts" />

class Foo {

}
```

**Comment by lavrton** [comment]
npm package of `Konva` will have typescript definition in next version.
If you install `Konva` from npm this code will work ok for typescript:
```javascript
import Konva from 'konva';
const Stage = new Konva.Stage({container: 'container'});
```

**Comment by deostroll** [comment]
@lavrton Can you leave this open until that is published over at npm...?

**Comment by stephenlaughton** [comment]
I am having no luck with the current definitions in vsCode/tsc>commonjs/Browserify. I keep on getting 
```
Uncaught TypeError: Cannot read property 'Stage' of undefined
```
When I try the simple example:
```
import Konva from "konva";

// first we need to create a stage
var stage = new Konva.Stage({
  container: 'container',   // id of container <div>
  width: 500,
  height: 500
});

// then create layer
var layer = new Konva.Layer();

// create our shape
var circle = new Konva.Circle({
  x: stage.getWidth() / 2,
  y: stage.getHeight() / 2,
  radius: 70,
  fill: 'red',
  stroke: 'black',
  strokeWidth: 4
});

// add the shape to the layer
layer.add(circle);

// add the layer to the stage
stage.add(layer);
```

**Comment by lavrton** [comment]
@stephenlaughton what version of `Konva` and `typescript` you are using?
I tested with latest `Konva` (v0.13.0) and last `typescript` version. All was fine.

**Comment by stephenlaughton** [comment]
@lavrton I am using the latest of both. Konva 0.13.0, VSCode 1.1.1, Typescript 1.8

What build system are you using to get your typescript bundled for the browser? I think that may be my issue because the Konva module is getting bundled incorrectly? I am using the latest Browserify. Maybe that is the issue?

Also is the .d.ts complete? I noticed some of the classes are empty. I can offer my assistance in fleshing that out with your direction if you want.

**Comment by lavrton** [comment]
So you have this error in runtime? Do you have any typescript compilation errors?

**Comment by lavrton** [comment]
I see the issue now too.

**Comment by stephenlaughton** [comment]
Yeah the error is at runtime. Any suggestions?

**Comment by lavrton** [comment]
For a hotfix you can use this ugly code (remove type annotations):
```
import * as konva from "konva";
const Konva : any = konva;
```


Then we need some fixes in `konva.d.ts` or in Konva code to support default exports.

**Comment by stephenlaughton** [comment]
@lavrton would you like some help on this? I am not too familiar with the codebase but with some direction I am sure I could help.

**Comment by stephenlaughton** [comment]
I actually got this working a bit better by modifying ```konva.d.ts```
By declaring the module as:
```typescript
declare module "Konva" {
...
    export class Animation {
        // instead of constructor(func: Function, layers?: Konva.Layer[]
        constructor(func: Function, layers?: Layer[]); 
        constructor(func: Function, layer?: Layer);
...
}
// and getting rid of the default export at the bottom
```
Now you can import like this:
```typescript
import * as Konva from "Konva";
```

**Comment by kargpfen** [comment]
Please use namspace instead of module and then declare "Konva" module at the end of file (like [d3.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/d3/d3.d.ts)). so it is better to use as module requirement (import) and/or reference only.

**Comment by lavrton** [comment]
Fixed and published in `0.14.0`. This syntax should work:
```javascript
import * as Konva from "konva";
```
